client/ts/graph_chart.ts

```diff
@@ -0,0 +1,64 @@
+import * as d3 from "d3";
+
+export function generateGraghChart(jsonUrl){
+
+    // let svg = d3.select('svg'),
+    //     width = document.getElementById("drawing-section").clientWidth,
+    //     height = document.getElementById("drawing-section").clientHeight,
+    //     nodeRadius = 10;
+    // svg.attr("width", width).attr("height", height);
+    let svg = d3.select('svg'),
+        height = +svg.attr("height"),
+        width = +svg.attr("width"),
+        nodeRadius = 8;
+
+    let simulation = d3.forceSimulation()
+        .force('charge', d3.forceManyBody())
+        .force('center', d3.forceCenter(width/2, height/2))
+        // .force('collide', d3.forceCollide().radius(nodeRadius*1.2))
+        .force('collide', d3.forceCollide())
+        .force('link', d3.forceLink().distance(80)); // distance sets the length of each link
+
+    d3.json(jsonUrl.value, function(error, data){
+        console.log(data);
+        let links = svg.append("g")
+            .selectAll("line")
+            .data(data.links)
+            .enter()
+            .append("line")
+            .attr("stroke", "#FFFFFF")
```

Comment:
Would be helpful to keep JS file with constants for the colors?

Reply:
good suggestions! I will create a color theme file